File: Day8/main.ts
function drawRace(indices: number[], length: number): string {
  const result: string[] = new Array(indices.length);

  for (let i = 0; i < indices.length; i++) {
    const index = indices[i];
    const shift: string = ' '.repeat(indices.length - i - 1);
    const lane: string = ` /${i + 1}`;
    const realIndex: number = ((index % length) + length) % length;
    const reindeerExistence: string = index === 0 ? '' : 'r';

    const raceLine: string = `${'~'.repeat(realIndex)}${reindeerExistence}`.padEnd(length, '~');
    result[i] = shift + raceLine + lane;
  }

  return result.join('\n');
}

File: Day8/main.py
def draw_race(indices, length):
    result = []
    length_minus_one = length - 1

    for i, index in enumerate(indices):
        shift = ' ' * (len(indices) - i - 1)
        lane = ' ' + f'/{i + 1}'
        real_index = (index % length + length) % length
        reindeer_existence = '' if index == 0 else 'r'

        race_line = ('~' * real_index) + reindeer_existence
        race_line = race_line.ljust(length, '~')
        result.append(shift + race_line + lane)

    return '\n'.join(result)
